app/logic/Mail/IMAP/IMAPFolder.ts

```diff
@@ -99,17 +99,19 @@ export class IMAPFolder extends Folder {
 
   /** Lists all messages in this folder.
    * But doesn't download their contents. @see downloadMessages() */
-  async listMessages() {
+  async listMessages(): Promise<ArrayColl<EMail>> {
     await this.readFolder();
     if (this.countTotal === 0) {
       return;
     }
     let { newMessages } = await this.fetchMessageList({ all: true }, {
         changedSince: this.lastModSeq,
       });
+    let newMsgs = new ArrayColl<IMAPEMail>(newMessages.subtract(this.messages as ArrayColl<IMAPEMail>));
```

Comment:
not merged. Will handle as part of #167

Reply:
Since I already had some TypeScript error fixes, I thought I'd just add the minimal fixes, but I didn't like the approach, which is why I went the other approach for #167.